Source/AwsCommonRuntimeKit/mqtt/Mqtt5Options.swift

```diff
@@ -547,42 +547,34 @@ public class NegotiatedSettings {
         self.clientId = clientId
     }
 
-    static func convertFromNative(_ from: UnsafePointer<aws_mqtt5_negotiated_settings>?) -> NegotiatedSettings? {
-
-        guard let from else {
-            return nil
-        }
-
-        let _negotiatedSettings = from.pointee
-        let negotiatedMaximumQos = QoS(_negotiatedSettings.maximum_qos)
-        let negotiatedSessionExpiryInterval: TimeInterval = TimeInterval(_negotiatedSettings.session_expiry_interval)
-        let negotiatedReceiveMaximumFromServer = _negotiatedSettings.receive_maximum_from_server
-        let negotiatedMaximumPacketSizeToServer = _negotiatedSettings.maximum_packet_size_to_server
-        let negotiatedTopicAliasMaximumToServer = _negotiatedSettings.topic_alias_maximum_to_server
-        let negotiatedTopicAliasMaximumToClient = _negotiatedSettings.topic_alias_maximum_to_client
-        let negotiatedServerKeepAlive: TimeInterval = TimeInterval(_negotiatedSettings.server_keep_alive)
-        let negotiatedRetainAvailable = _negotiatedSettings.retain_available
-        let negotiatedWildcardSubscriptionsAvailable = _negotiatedSettings.wildcard_subscriptions_available
-        let negotiatedSubscriptionIdentifiersAvailable = _negotiatedSettings.subscription_identifiers_available
-        let negotiatedSharedSubscriptionsAvailable = _negotiatedSettings.shared_subscriptions_available
-        let negotiatedRejoinedSession = _negotiatedSettings.rejoined_session
-        let negotiatedClientId = _negotiatedSettings.client_id_storage.toString()
-
-        let negotiatedSettings = NegotiatedSettings(
-            maximumQos: negotiatedMaximumQos,
-            sessionExpiryInterval: negotiatedSessionExpiryInterval,
-            receiveMaximumFromServer: negotiatedReceiveMaximumFromServer,
-            maximumPacketSizeToServer: negotiatedMaximumPacketSizeToServer,
-            topicAliasMaximumToServer: negotiatedTopicAliasMaximumToServer,
-            topicAliasMaximumToClient: negotiatedTopicAliasMaximumToClient,
-            serverKeepAlive: negotiatedServerKeepAlive,
-            retainAvailable: negotiatedRetainAvailable,
-            wildcardSubscriptionsAvailable: negotiatedWildcardSubscriptionsAvailable,
-            subscriptionIdentifiersAvailable: negotiatedSubscriptionIdentifiersAvailable,
-            sharedSubscriptionsAvailable: negotiatedSharedSubscriptionsAvailable,
-            rejoinedSession: negotiatedRejoinedSession,
-            clientId: negotiatedClientId)
-
-        return negotiatedSettings
+    internal convenience init(_ from: UnsafePointer<aws_mqtt5_negotiated_settings>){
```

Comment:
I think from is too generic, and since Swift allows us to have the same functions with different parameter names, it makes more sense to keep the parameter names explicit.

```suggestion
    internal convenience init(_ nativePointer: UnsafePointer<aws_mqtt5_negotiated_settings>){
```

Reply:
That's a good point. I was copy pasting these arguments but yea, we should definitely give them unique names.